pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/realtime/PinotLLCRealtimeSegmentManager.java

```diff
@@ -620,69 +622,76 @@ private void completeCommittingSegments() {
   }
 
   protected void completeCommittingSegments(String realtimeTableName) {
-    IdealState idealState = getTableIdealState(realtimeTableName);
-    Set<String> segmentNamesIS = idealState.getPartitionSet();
     List<ZNRecord> segmentMetadataList = getExistingSegmentMetadata(realtimeTableName);
     if (segmentMetadataList == null || segmentMetadataList.isEmpty()) {
       return;
     }
-    final List<LLCSegmentName> segmentNames = new ArrayList<>(segmentMetadataList.size());
+    final List<String> segmentIds = new ArrayList<>(segmentMetadataList.size());
 
     for (ZNRecord segment : segmentMetadataList) {
       if (SegmentName.isLowLevelConsumerSegmentName(segment.getId())) {
-        segmentNames.add(new LLCSegmentName(segment.getId()));
+        segmentIds.add(segment.getId());
       }
     }
 
-    if (segmentNames.isEmpty()) {
+    if (segmentIds.isEmpty()) {
       return;
     }
 
-    Collections.sort(segmentNames, Collections.reverseOrder());
+    completeCommittingSegments(realtimeTableName, segmentIds);
+  }
 
-    int curPartition = segmentNames.get(0).getPartitionId();  // Current kafka partition we are working on.
-    final int nSegments = segmentNames.size();
+  private void completeCommittingSegments(String realtimeTableName, Map<Integer, MinMaxPriorityQueue<LLCSegmentName>> partitionToLatestSegments) {
```

Comment:
Not sure why you bring this up, I think we use (and the standard libraries do as well) this pattern all over the place.